packages/analytics-node/src/node-client.ts

```diff
@@ -0,0 +1,226 @@
+import { AmplitudeCore, Destination } from '@amplitude/analytics-core';
+import { NodeConfig, NodeOptions } from '@amplitude/analytics-types';
+import { Context } from './plugins/context';
+import { useNodeConfig } from './config';
+
+export class AmplitudeNode extends AmplitudeCore<NodeConfig> {
+  async init(apiKey: string, userId?: string, options?: NodeOptions) {
+    const nodeOptions = useNodeConfig(apiKey, userId, {
+      ...options,
+      deviceId: options?.deviceId,
+      sessionId: options?.sessionId,
+      optOut: options?.optOut,
+    });
+
+    await super.init(undefined, undefined, nodeOptions);
+
+    await this.add(new Context());
+    await this.add(new Destination());
+  }
+
+  getUserId() {
+    return this.config.userId;
+  }
+
+  setUserId(userId: string) {
+    this.config.userId = userId;
+  }
+
+  getDeviceId() {
+    return this.config.deviceId;
+  }
+
+  setDeviceId(deviceId: string) {
+    this.config.deviceId = deviceId;
+  }
+
+  getSessionId() {
+    return this.config.sessionId;
+  }
+
+  setSessionId(sessionId: number) {
+    this.config.sessionId = sessionId;
+  }
```

Comment:
are these apis valuable in node sdk?

Reply:
Just check other SDKs, these APIs can be removed.

Reply:
generally we want it to look/feel like browser sdk but overall behavior should be pattered with other server-side sdks